### DISCUSSION TITLE
Gridded sampling and priority for dense spatial sampling

### DISCUSSION THREAD

**Discussion by ryabs07** [lululxvi/deepxde#1384] [Q&A]
Hi all,
Is gridded sampling possible in DeepXDE? I checked the source code and I could find only the following samplers: 
"pseudorandom", "LHS", "Halton", "Hammersley", "Sobol"

In any case that it is possible, can we set the distancing of sampling in spatial domain to be more dense than temporal domain?
In case it is not available yet, how do we add gridded sampling in DeepXDE?

Also, is it possible to give more weightage to spatial sampling than temporal sampling in the current version?
Any help is greatly appreciated, thank you :)

**Comment by ryabs07** [comment]
Is using the _anchors_ function the way to go for sampling specific points in geomtime domain?

for example:
```
x_start, x_end = 0, 1
x_first = np.linspace(x_start, 0.3, 100) #first segment of x with regular sampling
x_mid = np.linspace(0.3, 0.5, 300) #second segment of x with dense sampling
x_last = np.linspace(0.5, x_end, 100) #last segment of x with regular sampling
# Concatenate the two arrays to get the final x array
x = np.concatenate((x_first[0:-2], x_mid, x_last[1:-1]))

num_time_steps = 60 #Number of timesteps
t = np.linspace(time_start, time_end, num_time_steps)

#create a meshgrid in geomtime domain with desired sampling spacings
X, T = np.meshgrid(x,t) 
input = np.hstack((X.flatten()[:,None], T.flatten()[:,None]))
```
And I define the data for PINN as follows where I anchor the _input_ :

```
data = dde.data.TimePDE(
    geomtime, pde, [ic_h, ic_u, bc_l_h, bc_r_h, bc_l_u, bc_r_u], 
    num_domain=10000, num_boundary=1000, num_initial=1000, anchors=input
)
```

Please let me know if this approach is correct :) 
 Thank you

**Comment by lululxvi** [comment]
Use `uniform`, see https://deepxde.readthedocs.io/en/latest/modules/deepxde.data.html#deepxde.data.pde.PDE
